src/goto-symex/builtin_functions.cpp

```diff
@@ -748,277 +748,389 @@ void goto_symext::symex_va_arg(const expr2tc &lhs, const sideeffect2t &code)
-    code_function_call2t &mutable_funccall = to_code_function_call2t(newcall);
-    mutable_funccall.function =
-      symbol2tc(get_empty_type(), "c:@F@__memset_impl");
-    // Execute call
-    symex_function_call(newcall);
-    return;
-  };
+    for(int m = 0; m < 8; m++)
+    {
+      mask = shl2tc(type, mask, one);
+      mask = bitor2tc(type, mask, one);
+    }
+  }
 
-  // Skip if the operand is not zero. Because honestly, there's very little
-  // point.
-  cur_state->rename(value);
-  if(!is_constant_int2t(value) || to_constant_int2t(value).value != 0)
+  // Do the rest of the offset!
+  for(unsigned i = 0; i < offset; i++)
   {
-    bump_call();
-    return;
+    result = shl2tc(type, result, eight);
+    mask = shl2tc(type, mask, eight);
   }
 
-  // Work out what the ptr points at.
-  internal_deref_items.clear();
-  dereference2tc deref(get_empty_type(), ptr);
-  dereference(deref, dereferencet::INTERNAL);
+  mask = bitnot2tc(type, mask);
+  mask = bitand2tc(type, src, mask);
+  result = bitor2tc(type, result, mask);
 
-  // Work out here whether we can construct an assignment for each thing
-  // pointed at by the ptr.
-  cur_state->rename(size);
-  bool can_construct = true;
-  for(const auto &item : internal_deref_items)
-  {
-    const expr2tc &offs = item.offset;
+  auto simplified = result->simplify();
+  if(simplified)
+    return simplified;
 
-    int tmpsize;
-    try
-    {
-      tmpsize = type_byte_size(item.object->type).to_int64();
-    }
-    catch(array_type2t::dyn_sized_array_excp *e)
-    {
-      tmpsize = -1;
-    }
-    catch(array_type2t::inf_sized_array_excp *e)
-    {
-      tmpsize = -1;
-    }
+  return result;
+}
 
-    if(
-      is_constant_int2t(offs) && to_constant_int2t(offs).value == 0 &&
-      is_constant_int2t(size) && to_constant_int2t(size).value == tmpsize)
-    {
-      continue;
-    }
-    else if(
-      !is_constant_int2t(offs) && is_constant_int2t(size) &&
-      to_constant_int2t(size).value == tmpsize)
-    {
-      continue;
-    }
+inline expr2tc gen_value_by_byte(
+  const type2tc &type,
+  const expr2tc &src,
+  const expr2tc &value,
+  const size_t num_of_bytes,
+  const size_t offset)
+{
+  /**
+   * @brief Construct a new object, initializing it with the memset equivalent
+   *
+   * There are a few corner cases here:
+   * 
+   * 1 - Primitives: these are simple: just generate the byte_expression directly
+   * 2 - Arrays: these are ok: just keep generating byte_expression for each member
+   *        until a limit has arrived. Dynamic memory is dealt here.
+   * 3 - Structs/Union: these are the hardest as we have to take the alignment into
+   *        account when dealing with it. Hopefully the clang-frontend already give it
+   *        to us.
+   * 
+   */
+
+  // I am not sure if bitwise operations are valid for floats
+  if(is_floatbv_type(type) || is_fixedbv_type(type))
+    throw std::invalid_argument("This will not work for floats");
```

Comment:
can we get rid of the throws here? You can return expr2tc() and check the returned value on the caller.